gp-includes/warnings.php

```diff
@@ -815,4 +815,56 @@ private function links_without_url_and_placeholders_are_equal( string $original_
 
 		return $error;
 	}
+
+	/**
+	 * Adds a warning for added or removed spaces at the beginning or at the end of the translation.
+	 *
+	 * @since 4.0.0
+	 * @access public
+	 *
+	 * @param string $original    The original string.
+	 * @param string $translation The translated string.
+	 *
+	 * @return string|true True if check is OK, otherwise warning message.
+	 */
+	public function warning_unexpected_start_end_space( string $original, string $translation ) {
+		$original_start_spaces    = strlen( $original ) - strlen( ltrim( $original, ' ' ) );
```

Comment:
Where would you want to use such function?

Reply:
I am talking about the [strcmp()](https://www.php.net/manual/en/function.strcmp.php) function with multibyte support, so instead of 

`$original_start_spaces    = mb_strlen( $original ) - mb_strlen( ltrim( $original, ' ' ) );`

we can have something like this:

`$original_start_spaces    = mb_strcmp( $original, ltrim( $original, ' ' ) );`

But I think we don't have a function like this.

Reply:
I have updated the mb_strlen functions with some regex. Thanks to @akirk for the regexs.